### PR TITLE
mfgtools: add recipe

### PR DESCRIPTION
**Pull request by nefethael**
Signed-off-by: Vincent Prince <vincent.prince.fr@gmail.com>

### CODE REVIEW

File: recipes-devtools/mfgtools/mfgtoolscli_git.bb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+SRC_URI = " \
+    git://github.com/codeauroraforum/mfgtools;protocol=https \
+"
+
+PV = "1.0+git${SRCPV}"
+
+S = "${WORKDIR}/git"
+
+
+INSANE_SKIP_${PN} = "ldflags"
```
Comment (otavio) [inline recipes-devtools/mfgtools/mfgtoolscli_git.bb]: This goes after install as it is a packaging issue.

File: recipes-devtools/mfgtools/mfgtoolscli_git.bb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+do_compile(){
+    ${CXX} -I${STAGING_INCDIR}/libusb-1.0 -I${STAGING_INCDIR}/mfgtoolslib -fpermissive -Wno-write-strings ${S}/TestPrgm/mfgtoolCLI.cpp -o ${B}/mfgtoolcli -l:libMfgToolLibStatic.a -l:libfdt.a -lpthread -lusb-1.0
+}
+
+do_install(){
+    install -d ${D}${bindir}
+    install -m 0755 ${B}/mfgtoolcli ${D}${bindir}/mfgtoolcli
```
Comment (otavio) [inline recipes-devtools/mfgtools/mfgtoolscli_git.bb]: You can use install -Dm ... and drop line 24.

File: recipes-devtools/mfgtools/mfgtoolslib_git.bb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+S = "${WORKDIR}/git"
+
+inherit cmake
+OECMAKE_SOURCEPATH = "${S}/MfgToolLib"
+
+do_install(){
+    install -d ${D}${libdir}
+    install -m 0644 ${B}/libMfgToolLibStatic.a ${D}${libdir}/libMfgToolLibStatic.a
+    install -d ${D}${includedir}/mfgtoolslib
+    install -m 0644 ${S}/MfgToolLib/*.h ${D}${includedir}/mfgtoolslib/
```
Comment (otavio) [inline recipes-devtools/mfgtools/mfgtoolslib_git.bb]: The standard cmake install does not work? It'd be better to fix the CMakelists.txt

Also, why you don't use a single recipe to build the cli and lib?

**Comment by LocutusOfBorg** [comment]
Hello @otavio, are we good now? I see the recipe has been updated :)

**Comment by LocutusOfBorg** [comment]
oops, the split is still in place :/

**Comment by otavio** [comment]
Looking forward to review it again :-D

**Comment by LocutusOfBorg** [comment]
that code seems to have been removed upstream, I can't find the revision mentioned in the recipe... however google cache seems to show that file
https://webcache.googleusercontent.com/search?q=cache:4iEWUGsUODYJ:https://github.com/codeauroraforum/mfgtools/blob/master/TestPrgm/mfgtoolCLI.cpp+&cd=1&hl=en&ct=clnk&gl=it&client=ubuntu

**Comment by LocutusOfBorg** [comment]
branch=linux does the trick :)

**Comment by LocutusOfBorg** [comment]
This should do the trick:
```
From 9892b3c540718379501c4d20fb34207b8ce9e501 Mon Sep 17 00:00:00 2001
From: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Date: Mon, 21 May 2018 23:14:57 +0200
Subject: [PATCH 1/2] Simple-cmake

---
 CMakeLists.txt            |  3 +++
 MfgToolLib/CMakeLists.txt |  6 ++++--
 TestPrgm/CMakeLists.txt   | 12 ++++++++++++
 3 files changed, 19 insertions(+), 2 deletions(-)
 create mode 100644 CMakeLists.txt
 create mode 100644 TestPrgm/CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..56feb4b
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,3 @@
+cmake_minimum_required (VERSION 2.6)
+add_subdirectory(MfgToolLib)
+add_subdirectory(TestPrgm)
diff --git a/MfgToolLib/CMakeLists.txt b/MfgToolLib/CMakeLists.txt
index 00a242a..b8f75c1 100644
--- a/MfgToolLib/CMakeLists.txt
+++ b/MfgToolLib/CMakeLists.txt
@@ -18,8 +18,7 @@ cmake_minimum_required (VERSION 2.6)
 #    cmake_policy(SET CMP0015 NEW)
 #  endif(POLICY CMP0015)
 #endif(COMMAND cmake_policy)
-SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/")
-
+SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/MfgToolLib")
 
 #include_directories (${PROJECT_SOURCE_DIR}  {LIBUSB_1_INCLUDE_DIRS} )
 project( MfgToolLib )
@@ -183,4 +182,7 @@ set(SOURCES ${SOURCES_})
 
 add_library( MfgToolLib SHARED ${SOURCES} )
 add_library( MfgToolLibStatic STATIC ${SOURCES} )
+install( TARGETS MfgToolLib DESTINATION lib )
+set_property(TARGET MfgToolLib PROPERTY VERSION "0.0.0")
+set_property(TARGET MfgToolLib PROPERTY SOVERSION 0 )
 target_link_libraries(MfgToolLib ${LIBS})
diff --git a/TestPrgm/CMakeLists.txt b/TestPrgm/CMakeLists.txt
new file mode 100644
index 0000000..d85bc61
--- /dev/null
+++ b/TestPrgm/CMakeLists.txt
@@ -0,0 +1,12 @@
+cmake_minimum_required (VERSION 2.6)
+project( TestPrgm )
+SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/MfgToolLib")
+find_package(libusb-1.0 REQUIRED)
+include_directories(${LIBUSB_1_INCLUDE_DIRS})
+set(LIBS ${LIBS} ${LIBUSB_1_LIBRARIES} "fdt")
+add_definitions(-fpermissive)
+
+add_executable ( mfgtoolCLI mfgtoolCLI.cpp )
+include_directories( "${CMAKE_SOURCE_DIR}/MfgToolLib" )
+target_link_libraries( mfgtoolCLI MfgToolLib ${LIBS} )
+install( TARGETS mfgtoolCLI DESTINATION bin)
-- 
2.14.1
```
and
```
From 3d1fa6e6097b98bcb0efb0ab2155dc2e6cd7dfde Mon Sep 17 00:00:00 2001
From: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Date: Mon, 21 May 2018 23:44:49 +0200
Subject: [PATCH 2/2] Exit is int

---
 TestPrgm/mfgtoolCLI.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/TestPrgm/mfgtoolCLI.cpp b/TestPrgm/mfgtoolCLI.cpp
index 01a4a13..fafc070 100644
--- a/TestPrgm/mfgtoolCLI.cpp
+++ b/TestPrgm/mfgtoolCLI.cpp
@@ -58,7 +58,7 @@ std::string trim(std::string str)
 		last = 0.8 * (BAR_WIDTH);
 	return str.substr(first, (last-first+1));
 }
-void exitLibrary()
+int exitLibrary()
 {
 	int ret;
 	std::cout << std::endl << std::endl;
-- 
2.14.1
```

```
DESCRIPTION = "Freescale/NXP I.MX Chip image deploy tools library"
SECTION = "devel"
LICENSE = "BSD-3-Clause"
LIC_FILES_CHKSUM = "file://LICENSE;md5=76b1b3cba07c1020780276c11101dd5a"

DEPENDS = "libusb1 dtc"

SRCREV = "7ad03a087644f703afeb3b5204c9a9192466627f"
SRC_URI = " \
    git://github.com/codeauroraforum/mfgtools;protocol=https;branch=linux \
    file://0001-Fix-CXXFLAGS-in-CMakeList.patch \
    file://0001-Simple-cmake.patch \
    file://exit.patch \
"
PV = "1.0+git${SRCPV}"

S = "${WORKDIR}/git"

inherit cmake
```

File: recipes-devtools/mfgtools/mfgtoolslib_git.bb (hunk 3 above)
Comment (LocutusOfBorg) [reply recipes-devtools/mfgtools/mfgtoolslib_git.bb]: I fixed in a comment

**Comment by LocutusOfBorg** [comment]
Edit: updated also the first chunk to install the header files.

**Comment by otavio** [comment]
@LocutusOfBorg are going to send an updated PR?

**Comment by nefethael** [comment]
Sorry, I was in vacation, I'll update PR today. In a meanwhile I tried following [topic](https://community.nxp.com/thread/441563) steps , with imx_usb_loader and utp_com and I found it more conveniant in Linux.

**Comment by otavio** [comment]
that's indeed great. What about we get something like this well integrated so people can about reinventing the wheel?

**Comment by otavio** [comment]
@nefethael I think so; remember we have wic images which can be easily deployed in the device using bmaptool so we could consider this as an option for it.

**Comment by nefethael** [comment]
@LocutusOfBorg: no problem. In fact, I'm not very confident with cmake, so if you want to fix this PR, it would be nice. Also mfgtoolcli has a lot of issues, NXP is working on a new uuc and I switched to imx_usb_loader with otp-com, so I'm not sure this recipe is still needed.

**Comment by otavio** [comment]
> @LocutusOfBorg: no problem. In fact, I'm not very confident with cmake, so if you want to fix this PR, it would be nice. Also mfgtoolcli has a lot of issues, NXP is working on a new uuc and I switched to imx_usb_loader with otp-com, so I'm not sure this recipe is still needed.

Agreed; I think for now we can rely on utp-com. I am closing this for now.

**Pull request closed without merging**